File: docs/examples/py/src/data_ingestion_annotating.py
import time
import asyncio
from reduct import Client, Bucket


async def main():
    # Create a client instance, then get or create a bucket
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

        # Send a record  with labels and content type
        ts = time.time()
        await bucket.write(
            "py-example",
            b"Some binary data",
            ts,
            labels={"name": "example", "score": 0.9},
            content_type="text/plain",
        )


loop = asyncio.get_event_loop()
loop.run_until_complete(main())

File: docs/examples/go/src/data_ingestion_annotating.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Send a record with labels and content type
	err = bucket.BeginWrite(context.Background(), "py-example", &reduct.WriteOptions{
		Timestamp:   time.Now().UnixMicro(),
		Labels:      map[string]any{"name": "example", "score": 0.9},
		ContentType: "text/plain",
	}).Write([]byte("Some binary data"))

	if err != nil {
		panic(err)
	}
}
